doma-core/src/main/java/org/seasar/doma/jdbc/criteria/QueryDsl.java

```diff
@@ -35,6 +37,294 @@ public QueryDsl(Config config) {
+    Objects.requireNonNull(subQuery5);
+    Objects.requireNonNull(entityMetamodel6);
+    Objects.requireNonNull(subQuery6);
+    Objects.requireNonNull(entityMetamodel7);
+    Objects.requireNonNull(subQuery7);
+    Objects.requireNonNull(entityMetamodel8);
+    Objects.requireNonNull(subQuery8);
+    List<WithContext> withContexts =
+        List.of(
+            new WithContext(entityMetamodel1, subQuery1),
+            new WithContext(entityMetamodel2, subQuery2),
+            new WithContext(entityMetamodel3, subQuery3),
+            new WithContext(entityMetamodel4, subQuery4),
+            new WithContext(entityMetamodel5, subQuery5),
+            new WithContext(entityMetamodel6, subQuery6),
+            new WithContext(entityMetamodel7, subQuery7),
+            new WithContext(entityMetamodel8, subQuery8));
+    return with(withContexts);
+  }
+
+  public WithQueryDsl with(
+      EntityMetamodel<?> entityMetamodel1,
+      SetOperand<?> subQuery1,
+      EntityMetamodel<?> entityMetamodel2,
+      SetOperand<?> subQuery2,
+      EntityMetamodel<?> entityMetamodel3,
+      SetOperand<?> subQuery3,
+      EntityMetamodel<?> entityMetamodel4,
+      SetOperand<?> subQuery4,
+      EntityMetamodel<?> entityMetamodel5,
+      SetOperand<?> subQuery5,
+      EntityMetamodel<?> entityMetamodel6,
+      SetOperand<?> subQuery6,
+      EntityMetamodel<?> entityMetamodel7,
+      SetOperand<?> subQuery7,
+      EntityMetamodel<?> entityMetamodel8,
+      SetOperand<?> subQuery8,
+      EntityMetamodel<?> entityMetamodel9,
+      SetOperand<?> subQuery9) {
+    Objects.requireNonNull(entityMetamodel1);
+    Objects.requireNonNull(subQuery1);
+    Objects.requireNonNull(entityMetamodel2);
+    Objects.requireNonNull(subQuery2);
+    Objects.requireNonNull(entityMetamodel3);
+    Objects.requireNonNull(subQuery3);
+    Objects.requireNonNull(entityMetamodel4);
+    Objects.requireNonNull(subQuery4);
+    Objects.requireNonNull(entityMetamodel5);
+    Objects.requireNonNull(subQuery5);
+    Objects.requireNonNull(entityMetamodel6);
+    Objects.requireNonNull(subQuery6);
+    Objects.requireNonNull(entityMetamodel7);
+    Objects.requireNonNull(subQuery7);
+    Objects.requireNonNull(entityMetamodel8);
+    Objects.requireNonNull(subQuery8);
+    Objects.requireNonNull(entityMetamodel9);
+    Objects.requireNonNull(subQuery9);
+    List<WithContext> withContexts =
+        List.of(
+            new WithContext(entityMetamodel1, subQuery1),
+            new WithContext(entityMetamodel2, subQuery2),
+            new WithContext(entityMetamodel3, subQuery3),
+            new WithContext(entityMetamodel4, subQuery4),
+            new WithContext(entityMetamodel5, subQuery5),
+            new WithContext(entityMetamodel6, subQuery6),
+            new WithContext(entityMetamodel7, subQuery7),
+            new WithContext(entityMetamodel8, subQuery8),
+            new WithContext(entityMetamodel9, subQuery9));
+    return with(withContexts);
+  }
```

Comment:
It seems that overloads with a large number of parameters are inconvenient to use. How about replacing them with a method chaining approach using 'with'?

For example, you could add the following code to `WithQueryDsl`:

```java
public WithQueryDsl with(EntityMetamodel<?> entityMetamodel, SetOperand<?> subQuery) { ... }
```

The usage code would look like this:

```java
var query = dsl
    .with(dcCte1, dcCte1Query)
    .with(dcCte2, dcCte2Query)
    .from(e)
```

Reply:
ok, I fixed!

Reply:
Thanks!